worlds/kh1/__init__.py

```diff
@@ -0,0 +1,300 @@
+                    slot_data["keyblade_stats"] = slot_data["keyblade_stats"] + str(str_bonus) + "," + str(mp_bonus) + ","
+            slot_data["keyblade_stats"] = slot_data["keyblade_stats"][:-1]
+        if self.options.donald_death_link:
+            slot_data["donalddl"] = ""
+        if self.options.goofy_death_link:
+            slot_data["goofydl"] = ""
+        if self.options.keyblades_unlock_chests:
+            slot_data["chestslocked"] = ""
+        else:
+            slot_data["chestsunlocked"] = ""
+        if self.options.interact_in_battle:
+            slot_data["interactinbattle"] = ""
+        return slot_data
+    
+    def create_item(self, name: str) -> KH1Item:
+        data = item_table[name]
+        return KH1Item(name, data.classification, data.code, self.player)
+
+    def create_event(self, name: str) -> KH1Item:
+        data = event_item_table[name]
+        return KH1Item(name, data.classification, data.code, self.player)
+
+    def set_rules(self):
+        set_rules(self)
+
+    def create_regions(self):
+        create_regions(self.multiworld, self.player, self.options)
+    
+    def generate_early(self):
+        value_names = ["Reports to Open End of the World", "Reports to Open Final Rest Door", "Reports in Pool"]
+        initial_report_settings = [self.options.required_reports_eotw.value, self.options.required_reports_door.value, self.options.reports_in_pool.value]
+        self.change_numbers_of_reports_to_consider()
+        new_report_settings = [self.options.required_reports_eotw.value, self.options.required_reports_door.value, self.options.reports_in_pool.value]
+        for i in range(3):
+            if initial_report_settings[i] != new_report_settings[i]:
+                logging.info(f"{self.player_name}'s value {initial_report_settings[i]} for \"{value_names[i]}\" was invalid\n"
+                             f"Setting \"{value_names[i]}\" value to {new_report_settings[i]}")
+    
+    def change_numbers_of_reports_to_consider(self) -> None:
+        if self.options.end_of_the_world_unlock == "reports" and self.options.final_rest_door == "reports":
+            self.options.required_reports_eotw.value, self.options.required_reports_door.value, self.options.reports_in_pool.value = sorted(
+                [self.options.required_reports_eotw.value, self.options.required_reports_door.value, self.options.reports_in_pool.value])
+
+        elif self.options.end_of_the_world_unlock == "reports":
+            self.options.required_reports_eotw.value, self.options.reports_in_pool.value = sorted(
+                [self.options.required_reports_eotw.value, self.options.reports_in_pool.value])
+
+        elif self.options.final_rest_door == "reports":
+            self.options.required_reports_door.value, self.options.reports_in_pool.value = sorted(
+                [self.options.required_reports_door.value, self.options.reports_in_pool.value])
```

Comment:
This has to be one of the most interesting ways I've seen of solving this issue

*comment*